### ISSUE TITLE
ConnectionError

### ISSUE THREAD

**Issue by BeachGuy007** [ranaroussi/yfinance#2024]
### Describe bug

1 Failed download:
['UPRO']: ConnectionError(MaxRetryError('HTTPSConnectionPool(host=\'fc.yahoo.com\', port=443): Max retries exceeded with url: / (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x000001AD4DB41FD0>: Failed to resolve \'fc.yahoo.com\' ([Errno 11001] getaddrinfo failed)"))'))

This happens for all tickers.  UPRO is just an example.

### Simple code that reproduces your problem

import yfinance as yf
import pandas as pd

def fetch_data(tickers):
    ticker_data = {}
    for ticker in tickers:
        try:
            data = yf.download(ticker, 5d, 1m, threads=True)
            if not data.empty:
                ticker_data[ticker] = data['Close']
            else:
                ticker_data[ticker] = None
        except Exception as e:
            ticker_data[ticker] = None
    return ticker_data

def update_data(tickers):
    ticker_data = fetch_data(tickers)

    for ticker, data in ticker_data.items():
        if data is not None and len(data) > 0:
            latest_price = data.iloc[-1]
        else:
            latest_price = 'N/A'

    df = pd.DataFrame(data_rows, columns=["Ticker", "Price"])

def run_script():
    tickers = ["UPRO"]
    update_data(tickers)

if __name__ == "__main__":
    run_script()

### Debug log

1 Failed download:
['UPRO']: ConnectionError(MaxRetryError('HTTPSConnectionPool(host=\'fc.yahoo.com\', port=443): Max retries exceeded with url: / (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x000001AD4DB41FD0>: Failed to resolve \'fc.yahoo.com\' ([Errno 11001] getaddrinfo failed)"))'))

[*********************100%%**********************]  1 of 1 completed

1 Failed download:
['UPRO']: ConnectionError(MaxRetryError('HTTPSConnectionPool(host=\'fc.yahoo.com\', port=443): Max retries exceeded with url: / (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x000001AD4DC65A10>: Failed to resolve \'fc.yahoo.com\' ([Errno 11001] getaddrinfo failed)"))'))

### `yfinance` version

0.2.41

### Python version

3.11

### Operating system

WIN11

**Comment by BeachGuy007** [comment]
Mysteriously working now

**Comment by BeachGuy007** [comment]
It's on my end.  Am using "unbound" DNS in router.  Had Adblock turned on, that might have caused it.  Also I think the USB is glitchy.

**Comment by BeachGuy007** [comment]
close

**Comment by BeachGuy007** [comment]
Happened again this morning.  All quotes N/A

**Comment by BeachGuy007** [comment]
Working now.  I guess intermittent issue?

**Comment by BeachGuy007** [comment]
Issue with Yahoo servers?  Throttling?

**Comment by ValueRaider** [comment]
Are you spamming? https://github.com/ranaroussi/yfinance/discussions/1513

**Comment by BeachGuy007** [comment]
Ah no.  Real issue.

**Comment by BeachGuy007** [comment]
Now some quotes coming through.  I think it's yahoo servers(?).

**Comment by ValueRaider** [comment]
yfinance stores a cookie here: https://github.com/ranaroussi/yfinance#persistent-cache-store

Does deleting it help?

**Comment by BeachGuy007** [comment]
I am on WIN11 and there is no directory called py-yfinance here: Windows = C:/Users/<USER>/AppData/Local/py-yfinance

**Comment by BeachGuy007** [comment]
All quotes were N/A a second ago and just now flashed real quote.

**Comment by ValueRaider** [comment]
> I also don't have yf.set_tz_cache_location

... how did you check yfinance version?

**Comment by BeachGuy007** [comment]
PS C:\Users\Bill\Documents\Python> pip show yfinance
Name: yfinance
Version: 0.2.41
Summary: Download market data from Yahoo! Finance API
Home-page: https://github.com/ranaroussi/yfinance
Author: Ran Aroussi
Author-email: ran@aroussi.com
License: Apache
Location: C:\Users\Bill\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages
Requires: beautifulsoup4, frozendict, html5lib, lxml, multitasking, numpy, pandas, peewee, platformdirs, pytz, requests
Required-by:
PS C:\Users\Bill\Documents\Python>

**Comment by ValueRaider** [comment]
PIP can be wrong. Use the method in the new issue form.

**Comment by BeachGuy007** [comment]
Back to N/As again

**Comment by BeachGuy007** [comment]
import yfinance as yf ; print(yf.__version__)
0.2.41

**Comment by ValueRaider** [comment]
Regardless, if you don't have `yf.set_tz_cache_location` then your version must be almost 1-year old, or corrupt. https://github.com/ranaroussi/yfinance/commit/59af19d84c73e026730ac2c48310316011fe7efd

**Comment by BeachGuy007** [comment]
I had the following but chatGPT said:
Removed Invalid Cache Path:
The yf.set_tz_cache_location call was removed because it is not a valid method for managing cache in yfinance.

import datetime
import yfinance as yf
yf.set_tz_cache_location("C:\\Users\\Bill\Documents\\Python\\yfinance\\py-yfinance")

**Comment by BeachGuy007** [comment]
Ok I uninstalled yfinance rebooted and reinstalled yfinance.  Quotes seem to be working now.  I'll let you know if something happens.  Thanks for all your help!

pip uninstall yfinance
pip install yfinance

**Comment by BeachGuy007** [comment]
New error even after uninstalling/reinstalling.

**Comment by BeachGuy007** [comment]
1 Failed download:
2024-08-21 08:17:14,639:ERROR:['UPRO']: YFPricesMissingError('$%ticker%: possibly delisted; No price data found (period=1d)')

**Comment by BeachGuy007** [comment]
I wrote some reconnection python code and it seems to be working with that.

**Comment by bastienjalbert** [comment]
I don't think my root cause was the same as your but if you use **custom DNS server** (for example **pihole**) fc.yahoo.com could be blocked. So ensure you can open and connect to the url from the device and you may check your dns.

**Comment by BeachGuy007** [comment]
Yes, I use unbound and when I connect to fc.yahoo.com it redirects to https://www.yahoo.com/.  But I don't think unbound is the problem as my scripts work fine on Linux OS.

**Comment by ernstleierzopf** [comment]
> I don't think my root cause was the same as your but if you use **custom DNS server** (for example **pihole**) fc.yahoo.com could be blocked. So ensure you can open and connect to the url from the device and you may check your dns.

you are a genius, thank you very much!
